pkg/controller/errors.go

```diff
@@ -0,0 +1,19 @@
+package controller
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+)
+
+// errLogAndWrap is a small utility function to reduce code and be able to directly
+// return errors, but wrap them and log them at the same time.
+//
+// The function currently capitalizes the first letter as well.
+// message is expected to be NOT empty.
```

Comment:
Expected but not checked? In case of an empty message (a) a panic could be raised (iiirks) or (b) at least the filename and line number of the caller could be retrieved and the message could be "No error information given at file:line".

Also in my eyes it's more an information than a message.